script.js

```diff
@@ -1,4 +1,456 @@
+  } else if (gameStatus == "player selection") {
+    if (Number.isNaN(Number(input)) == true || input < 2 || input == "") {
+      returnStatement = `Wrong input. Please enter a number greater or equal to 2.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  } else if (gameStatus == "dice selection") {
+    if (Number.isNaN(Number(input)) == true || input == 0 || input == "") {
+      returnStatement = `Wrong input. Please enter a number greater than 0.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  } else if (gameStatus == "mode selection") {
+    if (
+      !(input.toUpperCase() == "KNOCKOUT" || input.toUpperCase() == "ORIGINAL")
+    ) {
+      returnStatement = `Wrong input. Please enter only "Original" or "Knockout" to start the game.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  }
+};
+
+// Function to generate random dice number
+var rollDice = function () {
+  var randomDiceNum = Math.floor(Math.random() * 6) + 1;
+  return randomDiceNum;
+};
+
+// Function to add random dice number to array
+var generateDiceResult = function () {
+  var counter = 0;
+  rollDiceResult = [];
+  while (counter < numOfDice) {
+    rollDiceResult.push(rollDice());
+    counter += 1;
+  }
+  return rollDiceResult;
+};
+
+// Function to copy an array into another array
+
+var copyArray = function (arrayToCopy) {
+  var newArray = [];
+  for (var counter = 0; counter < arrayToCopy.length; counter += 1) {
+    newArray.push(arrayToCopy[counter]);
+  }
+  return newArray;
+};
+
+// Function to auto concatenate dice numbers
+
+var autoConcatenateDiceNum = function (inputArray) {
+  var tempArray = copyArray(inputArray);
+  var sortedTempArray = tempArray.sort();
+  var concatenateString = "";
+
+  if (gameVersion == "normal") {
+    for (var counter = sortedTempArray.length - 1; counter >= 0; counter -= 1) {
+      var diceNumber = sortedTempArray[counter];
+      concatenateString = concatenateString + diceNumber.toString();
+    }
+
+    concatenateNum = Number(concatenateString);
+
+    return concatenateNum;
+  } else if (gameVersion == "lowestCombined") {
+    for (var counter = 0; counter < sortedTempArray.length; counter += 1) {
+      var diceNumber = sortedTempArray[counter];
+      concatenateString = concatenateString + diceNumber.toString();
+    }
+    concatenateNum = Number(concatenateString);
+
+    return concatenateNum;
+  }
+};
+
+// Function to check who wins for the round
+var checkWhoWins = function (array) {
+  var resultStatement = "";
+  var maxScoreIndex = 0;
+  var duplicateCount = 0;
+  var currentScore = array[0].playerConcatenateNum;
+
+  // Find the maximum score
+  if (gameVersion == "normal") {
+    for (var counter = 0; counter < array.length; counter += 1) {
+      if (array[counter].playerConcatenateNum > currentScore) {
+        currentScore = array[counter].playerConcatenateNum;
+        maxScoreIndex = counter;
+      }
+    }
```

Comment:
Good job using an array to cycle through the possible scores and matching them up against each other!